File: Programme_Python/exec/exo-1.py
import sys

if len(sys.argv) != 3:
    print(f"Usage : {sys.argv[0]} nombre1 nombre2")
    sys.exit(1)

# Recuperation en int
try:
    a = int(sys.argv[1])
    b = int(sys.argv[2])
except ValueError:
    print("Les arguments doivent être des nombres entiers.")
    sys.exit(1)

# Calcul et afficher la somme
somme = a + b
print(f"La somme de {a} et {b} est : {somme}")

File: Programme_C/chapitre-3/exo-1.c

/*Exercice 3.1 : Écrire un programme qui prend deux arguments en ligne de commande en
supposant qu ce sont des nombres entiers, et qui affiche l’addition de ces deux nombres.*/

#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
    if (argc != 3)
    {
        printf("Usage : %s nombre1 nombre2\n", argv[0]);
        return 1;
    }

    int a = atoi(argv[1]);
    int b = atoi(argv[2]);

    int somme = a + b;

    printf("La somme de %d et %d est : %d\n", a, b, somme);

    return 0;
}